test/distributed/org/apache/cassandra/distributed/shared/ClusterUtils.java

```diff
@@ -1510,5 +1573,67 @@ public static TableId tableId(Cluster cluster, String ks, String table)
         String str = cluster.getFirstRunningInstance().callOnInstance(() -> Schema.instance.getKeyspaceInstance(ks).getColumnFamilyStore(table).getTableId().toString());
         return TableId.fromUUID(UUID.fromString(str));
     }
+
+    public static void awaitInPeers(Cluster cluster, int[] nodes, IInstance expectedInPeers)
```

Comment:
> 10) fixed a race condition bug in Harry where the TokenPlacementModel could see a partial row causing NPEs down the line

This is a better fix than the one in harry, but since harry is used in other contexts I kept the logic there...

TCM async updates the peers tables, and the updates are not atomic and the caller can see partial state... this logic makes sure that the expected peer is *complete* in the expected nodes... this resolves the infrequent NPEs in Harry (which the retries at that layer are trying to do as well)